Page builder adjustments

aachavarria opened the pull request:
adding snackbar to PreviewConcierge.tsx;
adding desktop upload assets progress mask

aachavarria committed:
adding default anchorOrigin to snackbarProvider
using snackbar provider to show the operations feedback

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into pageBuilderAdjustments

aachavarria committed:
adding constants DESKTOP_ASSET_UPLOAD_PROGRESS;
creating AssetUploaderMask;
changes to tracking uploading files;
upload progress support;

rart commented on ui/app/src/components/CrafterCMSNextBridge.tsx:
```diff
@@ -82,7 +82,7 @@ function CrafterCMSNextBridge(props: PropsWithChildren<{ isLegacy?: boolean }>)
       <RawIntlProvider value={intl}>
         <ThemeProvider theme={theme}>
-          <SnackbarProvider maxSnack={5}>
+          <SnackbarProvider maxSnack={5} anchorOrigin={{ vertical: 'top', horizontal: 'right' }}>
```
Are these just defaults? Consumers may override?

aachavarria replied:
yes

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -410,20 +411,15 @@ function beginGuestDetection(setSnack): Subscription {
+      >
+        Learn More
+      </Button>
     });
```
Format

rart commented on ui/guest/src/util.tsx:
```diff
@@ -460,6 +466,15 @@ export function retrieveProperty(object: object, prop: string): any {
+    }
+    return next;
+  }, {});
+}
```
This is sort of equivalent to reversePluckProps. 

A deleteProperty should be more like
```typescript
export function deleteProperty<T, P extends keyof T>(object: T, prop: P): Omit<T, P> {
  delete object[prop];
  return object;
}
```

Notice it removes the prop from the source object.

rart replied:
Add to app's object.ts too.

rart reviewed (changes requested):
Is this still on draft stage on purpose — still being worked?

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into pageBuilderAdjustments

aachavarria committed:
adding missing dependencies,
adding DESKTOP_ASSET_DROP user feedback

aachavarria committed:
formatting issues;
changes on deleteProperty;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into pageBuilderAdjustments

 Conflicts:
	ui/app/src/components/CrafterCMSNextBridge.tsx

aachavarria committed:
missing dependencies;

rart commented on ui/app/src/components/BulkUpload.tsx:
```diff
@@ -678,7 +671,8 @@ export default function BulkUpload(props: BulkUploadProps) {
         title={formatMessage(translations.title)}
         subtitle={formatMessage(translations.subtitle)}
-        onClose={dropZoneStatus.status === 'uploading' ? onMinimized : () => onClose(dropZoneStatus)}
+        //onClose={dropZoneStatus.status === 'uploading' ? onMinimized : () => onClose(dropZoneStatus)}
```
x

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -286,18 +287,29 @@ export function PreviewConcierge(props: any) {
+              enqueueSnackbar('Asset Upload failed.');
             },
             () => {
+              console.log('finished');
```
x

aachavarria committed:
adding pulse animation, and autohide default to 3000

aachavarria committed:
Feedback

rart committed:
Update format & default duration

aachavarria committed:
Fixing issue when uploading a image and the prev images has not finished;
adding cancelAll on uppy.unsubscribe;
adding DESKTOP_ASSET_UPLOAD_STARTED;
Guest.tsx: adding elementZoneId on the payload;

aachavarria committed:
Merge remote-tracking branch 'aaron/pageBuilderAdjustments' into pageBuilderAdjustments
